app/controllers/serviceLearning/routes.py

```diff
@@ -34,33 +34,75 @@ def serviceCourseManagement(username=None):
         return redirect(url_for('main.events', selectedTerm=g.current_term))
 
 @serviceLearning_bp.route('/serviceLearning/viewProposal/<courseID>', methods=['GET'])
+def slcViewProposal(courseID):
```

Comment:
This function is not necessary because the view proposal functionality has already been handled by 'slcNewProposal.js' and 'slcQuestionnaire.html'

To direct the page to viewProposal when s1 click edit on approved proposal, you could change a condition in the 'changAction' function in the 'slcNewProposal.js'. 

Please add 
`    if g.current_user.isStudent:
        abort(403)`

in the `slcEditProposal` to make sure students cannot access the view and edit proposal page.